packages/go_router/example/lib/stateful_nested_navigation.dart

```diff
@@ -0,0 +1,291 @@
+        /// pageProvider function in addition to the builder, for example:
+        // pageProvider:
+        //     (BuildContext context, GoRouterState state, Widget statefulShell) {
+        //   return NoTransitionPage<dynamic>(child: statefulShell);
+        // },
+
+        /// If you need to create a custom container for the branch routes, to
+        /// for instance setup animations, you can implement your builder
+        /// something like this:
+        // builder:
+        //     (BuildContext context, GoRouterState state, Widget ignoringThis) {
+        //   final StatefulShellRouteState shellRouteState =
+        //       StatefulShellRoute.of(context);
+        //   return ScaffoldWithNavBar(
+        //       body: _AnimatedRouteBranchContainer(
+        //     currentIndex: shellRouteState.index,
+        //     navigators: shellRouteState.navigators,
+        //   ));
+        // },
+      ),
+    ],
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      routerConfig: _router,
+    );
+  }
+}
+
+/// Builds the "shell" for the app by building a Scaffold with a
+/// BottomNavigationBar, where [child] is placed in the body of the Scaffold.
+class ScaffoldWithNavBar extends StatelessWidget {
+  /// Constructs an [ScaffoldWithNavBar].
+  const ScaffoldWithNavBar({
+    required this.body,
+    Key? key,
+  }) : super(key: key ?? const ValueKey<String>('ScaffoldWithNavBar'));
+
+  /// Body, i.e. the index stack
+  final Widget body;
+
+  @override
+  Widget build(BuildContext context) {
+    final StatefulShellRouteState shellState = StatefulShellRoute.of(context);
+    return Scaffold(
+      body: body,
+      bottomNavigationBar: BottomNavigationBar(
+        items: const <BottomNavigationBarItem>[
+          BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Section A'),
+          BottomNavigationBarItem(
+              icon: Icon(Icons.settings), label: 'Section B'),
+        ],
+        currentIndex: shellState.index,
+        onTap: (int tappedIndex) => _onItemTapped(
+          context,
+          shellState.navigationBranchState[tappedIndex],
+        ),
+      ),
+    );
+  }
+
+  void _onItemTapped(BuildContext context, ShellRouteBranchState routeState) {
+    GoRouter.of(context).go(routeState.location);
+  }
+}
+
+/// Widget for the root/initial pages in the bottom navigation bar.
+class RootScreen extends StatelessWidget {
+  /// Creates a RootScreen
+  const RootScreen(
+      {required this.label,
+      required this.detailsPath,
+      this.detailsPath2,
+      Key? key})
+      : super(key: key);
+
+  /// The label
+  final String label;
+
+  /// The path to the detail page
+  final String detailsPath;
+
+  /// The path to another detail page
+  final String? detailsPath2;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```

Comment:
@tolo 

Thanks for the PR 🙏 

Scaffold is nested with this, and it is not recommended: https://github.com/flutter/packages/blob/ee9bde9dc50f8a5145cdcef50cf3780cae8c4960/packages/go_router/example/lib/stateful_nested_navigation.dart#L141

<img width="572" alt="Screen Shot 2022-10-12 at 10 04 38" src="https://user-images.githubusercontent.com/1255062/195225836-58d25b51-59bc-44ce-9881-47d1be7c75ab.png">

I think this Scaffold should be removed.

> ## Troubleshooting
> ### Nested Scaffolds
> The Scaffold is designed to be a top level container for a [MaterialApp](https://api.flutter.dev/flutter/material/material/MaterialApp-class.html). This means that adding a Scaffold to each route on a Material app will provide the app with Material's basic visual layout structure.
> 
> It is typically not necessary to nest Scaffolds. For example, in a tabbed UI, where the [bottomNavigationBar](https://api.flutter.dev/flutter/material/material/Scaffold/bottomNavigationBar.html) is a [TabBar](https://api.flutter.dev/flutter/material/material/TabBar-class.html) and the body is a [TabBarView](https://api.flutter.dev/flutter/material/material/TabBarView-class.html), you might be tempted to make each tab bar view a scaffold with a differently titled AppBar. Rather, it would be better to add a listener to the [TabController](https://api.flutter.dev/flutter/material/material/TabController-class.html) that updates the AppBar
> 
> [Scaffold class - material library - Dart API](https://api.flutter.dev/flutter/material/Scaffold-class.html)